app/src/main/java/com/nextcloud/talk/location/GeocodingActivity.kt

```diff
@@ -108,16 +114,17 @@ class GeocodingActivity :
             Log.e(TAG, "search string that was passed to GeocodingController was null or empty")
         }
 
-        binding.geocodingResults.onItemClickListener = AdapterView.OnItemClickListener { parent, view, position, id ->
-            val address: Address = adapter.getItem(position) as Address
-            val geocodingResult = GeocodingResult(address.latitude, address.longitude, address.displayName)
-
-            val intent = Intent(this, LocationPickerActivity::class.java)
-            intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP)
-            intent.putExtra(BundleKeys.KEY_ROOM_TOKEN, roomToken)
-            intent.putExtra(BundleKeys.KEY_GEOCODING_RESULT, geocodingResult)
-            startActivity(intent)
-        }
+        adapter.setOnItemClickListener(object : GeocodingAdapter.OnItemClickListener {
+            override fun onItemClick(position: Int) {
+                val address: Address = adapter.getItem(position) as Address
+                val geocodingResult = GeocodingResult(address.latitude, address.longitude, address.displayName)
+                val intent = Intent(this@GeocodingActivity, LocationPickerActivity::class.java)
+                intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP)
+                intent.putExtra(BundleKeys.KEY_ROOM_TOKEN, roomToken)
+                intent.putExtra(BundleKeys.KEY_GEOCODING_RESULT, geocodingResult)
+                startActivity(intent)
+            }
+        })
```

Comment:
``` 
adapter.setOnItemClickListener(object : GeocodingAdapter.OnItemClickListener {
            override fun onItemClick(position: Int) {
                val address: Address = adapter.getItem(position) as Address
                val geocodingResult = GeocodingResult(address.latitude, address.longitude, address.displayName)
                val intent = Intent(this@GeocodingActivity, LocationPickerActivity::class.java)
                intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP)
                intent.putExtra(BundleKeys.KEY_ROOM_TOKEN, roomToken)
                intent.putExtra(BundleKeys.KEY_GEOCODING_RESULT, geocodingResult)
                startActivity(intent)
            }
      })
```

move this into the `initAdapter` function, before the `binding.geocodingResults.adapter = adapter`. Currently, the adapter is not being updated after the ItemClicker is set up.

Reply:
Nice catch. 
Thank you @rapterjet2004 for the fix 😁

Reply:
Resolved the issue